core-js/src/main/javascript/cdf-legacy/queries/coreQueries.js

```diff
@@ -114,25 +115,55 @@
       if(typeof this.getOption('successCallback') != 'function') {
         throw 'QueryNotInitialized';
       }
-      var url = this.getOption('url'),
-          callback = (successCallback ? successCallback : this.getOption('successCallback')),
-          errorCallback = (errorCallback ? errorCallback : this.getOption('errorCallback')),
-          queryDefinition = this.buildQueryDefinition();
 
-      var settings = _.extend({}, this.getOption('ajaxOptions'), {
+      var successHandler = successCallback ? successCallback : this.getOption('successCallback');
+      var errorHandler = errorCallback ? errorCallback : _.bind(this.getOption('errorCallback'), this);
+      var queryDefinition = this.buildQueryDefinition();
+
+      var isFirstTime = true;
+
+      if(this.getOption('pushEnabled')) {
+        var myself = this;
+
+        this.websocket = new WebSocket(this.getOption('websocketUrl'), 'JSON-CDA-query');
+
+        this.websocket.onopen = function (event) {
+          myself.websocket.send(JSON.stringify(queryDefinition));
+        };
+
+        this.websocket.onmessage = function (event) {
+          var isDataPush = !isFirstTime;
+          isFirstTime = false;
+
+          myself.getSuccessHandler(function (data) {
+            return successHandler(data, isDataPush);
+          })(JSON.parse(event.data));
```

Comment:
`, isDataPush` should be moved to after `JSON.parse(event.data)`:

```js
myself.getSuccessHandler(successHandler)(JSON.parse(event.data), isDataPush));
```

Which then prompts us to move `myself.getSuccessHandler(successHandler)` tranformation directly to the variable `successHandler` above.

Reply:
Ignore.